website/content/docs/secrets/databases/redis.mdx

```diff
@@ -0,0 +1,126 @@
+description: |-
+  Redis is one of the supported plugins for the database secrets engine.
+  This plugin generates database credentials dynamically based on configured
+  roles for the Redis database, and also supports [Static Roles](https://developer.hashicorp.com/vault/docs/secrets/databases#static-roles).
+---
+
+# Redis Database Secrets Engine
+
+Redis is one of the supported plugins for the database secrets engine. This
+plugin generates database credentials dynamically based on configured roles for
+the Redis database.
+
+See the [database secrets engine](/docs/secrets/databases) docs for
+more information about setting up the database secrets engine.
+
+## Capabilities
+
+| Plugin Name                 | Root Credential Rotation | Dynamic Roles | Static Roles | Username Customization |
+| --------------------------- | ------------------------ | ------------- | ------------ | ---------------------- |
+| `redis-database-plugin`     | Yes                      | Yes           | Yes          | No                     |
+
+## Setup
+
+1.  Enable the database secrets engine if it is not already enabled:
+
+    ```shell-session
+    $ vault secrets enable database
+    Success! Enabled the database secrets engine at: database/
+    ```
+
+    By default, the secrets engine will enable at the name of the engine. To
+    enable the secrets engine at a different path, use the `-path` argument.
+
+1.  Configure Vault with the proper plugin and connection configuration:
+
+    ```shell-session
+    $ vault write database/config/my-redis-database \
+      plugin_name="redis-database-plugin" \
+      host="localhost" \
+      port=6379 \
+      tls=true \
+      ca_cert="$CACERT"
+      username="user" \
+      password="pass" \
+      allowed_roles="my-*-role"
+    ```
+
+1.  You should consider rotating the admin password. Note that if you do, the
+    new password will never be made available through Vault, so you should
+    create a Vault-specific database admin user for this.
+
+    ```shell-session
+    vault write -force database/rotate-root/my-redis-database
+    ```
+
+## Usage
+
+After the secrets engine is configured, configure dynamic and static roles
+to enable generating credentials.
+
+### Dynamic roles
+
+1.  Configure a dynamic role that maps a name in Vault to a JSON string
+    containing the Redis ACL rules which are documented [here](https://redis.io/commands/acl-cat) or in the output
+    of the `ACL CAT` redis command.
+
+    ```shell-session
+    $ vault write database/roles/my-dynamic-role \
+        db_name="my-redis-database" \
+        creation_statements='["+@admin"]' \
+        default_ttl="5m" \
+        max_ttl="1h"
+    Success! Data written to: database/roles/my-dynamic-role
+    ```
+
+Note that if a creation_statement is not provided the user account will
+default to a read only user, `'["~*", "+@read"]'` that can read any key.
+
+1.  Generate a new credential by reading from the `/creds` endpoint with the name
```

Comment:
Maybe this should be more specific and say "generate a new set of credentials by..." since it makes both a username and password?

Reply:
Thank you for suggestions and reviewing! 
I think I will leave this one as is, because I just checked all of the other database secrets engine documents, and every single one of them is formulated like this. So, to stay consistent I think that we should do the same for this one? Would you agree? @robmonte

Reply:
I think _ideally_ the others would also be updated to reflect this, but it's such a trivial detail that I don't think it matters much. And updating the wording of other pages is outside of the scope here.

FWIW, I don't think things always need to stay consistent when the change is an improvement, but for this minor wording difference I am happy to leave it this way if you prefer it to be. 👍

Reply:
I think that you are right, eventually we can update all of them. I went ahead and updated it to say "a new set of credentials by". Thank you so much!